### PR TITLE
fix dataset page bug, framework add transformers dependency

### PR DESCRIPTION
**Pull request by liuyhwangyh**
1. Fix dataset file download pagination bug.
2. Add transformers to framework.
3. Pillow to framework.
4. Command line add --token parameter

### CODE REVIEW

File: modelscope/cli/cli.py [hunk 1]
```diff
@@ -17,6 +18,8 @@
 def run_cmd():
     parser = argparse.ArgumentParser(
         'ModelScope Command Line tool', usage='modelscope <command> [<args>]')
+    parser.add_argument(
+        '--token', default=None, help='Specify modelscope token.')
```
Comment (yingdachen) [inline modelscope/cli/cli.py:22]: Specify ModelScope SDK token.

File: modelscope/hub/snapshot_download.py [hunk 2]
```diff
@@ -56,6 +60,10 @@ def snapshot_download(
         allow_file_pattern (`str` or `List`, *optional*, default to `None`):
             Any file pattern to be downloading, like exact file names or file extensions.
         local_dir (str, optional): Specific local directory path to which the file will be downloaded.
+        allow_patterns (`str` or `List`, *optional*, default to `None`):
```
Comment (yingdachen) [inline modelscope/hub/snapshot_download.py:63]: 加一句For hugging-face compatability.

File: modelscope/hub/snapshot_download.py [hunk 3]
```diff
@@ -56,6 +60,10 @@ def snapshot_download(
...
+            If provided, only files matching at least one pattern are downloaded, priority over allow_file_pattern.
+        ignore_patterns (`str` or `List`, *optional*, default to `None`):
+            If provided, files matching any of the patterns are not downloaded, priority over ignore_file_pattern.
```
Comment (yingdachen) [inline modelscope/hub/snapshot_download.py:66]: 同上

File: modelscope/hub/snapshot_download.py [hunk 4]
```diff
@@ -120,6 +136,10 @@ def dataset_snapshot_download(
             Use regression is deprecated.
         allow_file_pattern (`str` or `List`, *optional*, default to `None`):
             Any file pattern to be downloading, like exact file names or file extensions.
+        allow_patterns (`str` or `List`, *optional*, default to `None`):
```
Comment (yingdachen) [inline modelscope/hub/snapshot_download.py:139]: 加一句For hugging-face compatability.

File: modelscope/hub/snapshot_download.py [hunk 5]
```diff
@@ -120,6 +136,10 @@ def dataset_snapshot_download(
...
+            If provided, only files matching at least one pattern are downloaded, priority over allow_file_pattern.
+        ignore_patterns (`str` or `List`, *optional*, default to `None`):
```
Comment (yingdachen) [inline modelscope/hub/snapshot_download.py:141]: 加一句For hugging-face compatability.

File: modelscope/cli/cli.py (hunk 1 above)
Comment (liuyhwangyh) [reply modelscope/cli/cli.py]: fixed

File: modelscope/hub/snapshot_download.py (hunk 2 above)
Comment (liuyhwangyh) [reply modelscope/hub/snapshot_download.py:63]: fixed

File: modelscope/hub/snapshot_download.py (hunk 3 above)
Comment (liuyhwangyh) [reply modelscope/hub/snapshot_download.py:67]: fixed

File: modelscope/hub/snapshot_download.py (hunk 4 above)
Comment (liuyhwangyh) [reply modelscope/hub/snapshot_download.py:141]: fixed

File: modelscope/hub/snapshot_download.py (hunk 5 above)
Comment (liuyhwangyh) [reply modelscope/hub/snapshot_download.py:144]: fixed

**Review by yingdachen — APPROVED** [review]
LGTM

**Review by yingdachen — APPROVED** [review]

**Pull request merged**